src/test/java/org/mamute/infra/NormalizerBrutalTest.java

```diff
@@ -27,6 +32,9 @@ public void should_remove_question_marks_and_bangs() {
 		String text = "como faz para normalizar uma string?!";
 		String slug = NormalizerBrutal.toSlug(text);
 		assertEquals("como-faz-para-normalizar-uma-string", slug);
+
+		String slugEncoded = NormalizerBrutal.toSlug(text, true);
+		assertEquals("como-faz-para-normalizar-uma-string%3f%21", slugEncoded);
```

Comment:
Apologies for jumping in. The current pull request is not usable with Asian languages. I would think just using URL encoding is the simplest and covers the most cases for the least work particularly as browsers now [handle the display of Unicode urls](http://stackoverflow.com/questions/2742852/unicode-characters-in-urls) better. There are a few characters such as "?" that are still displayed as % hex code in the url input box but there are good security reasons for that. Realizing this is a question site and most titles will end in a question mark that would show up as %3F  perhaps the only time a character should be stripped is when the title ends in a question mark.